src/Jobs/PtpJob.php

```diff
@@ -0,0 +1,324 @@
+use Exception;
+use File;
+use FileCache;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Bus\Batchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Throwable;
+
+class PtpJob extends BaseJob implements ShouldQueue
+{
+    use Batchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     *
+     * @var string $tmpInputFile File where the input data for the Python script will be kept
+     * @var string $tmpImageInputFile File where the image input data for the Python script will be kept
+     * @var int $insertChunkSize Number of annotations to be inserted per chunk
+     *
+     */
+    private string $tmpInputFile;
+    private string $tmpImageInputFile;
+    public static int $insertChunkSize = 5000;
+
+    /**
+     * Job used for converting Point annotations to Polygons
+     *
+     * @param int $volumeId Id of the volume for the PTP Job
+     * @param string $volumeName Name of the volume for the PTP Job
+     * @param string $outputFile File that will contain the resulting conversions
+     * @param string $inputFile Input JSON file containing the annotations to convert
+     * @param User $user User starting the PtpJob
+     * @param string $id Uuid associated to the job
+     *
+     */
+    public function __construct(
+        public int $volumeId,
+        public string $volumeName,
+        public string $inputFile,
+        public string $outputFile,
+        public User $user,
+        public string $id,
+    )
+    {
+        if (config('ptp.ptp_fail_construct') && config('ptp.ptp_storage_disk') == 'test') {
+            throw new Exception("Error!");
+        }
+
+        $this->volumeId = $volumeId;
+        $this->volumeName = $volumeName;
+        $this->outputFile = config('ptp.temp_dir').'/'.$outputFile;
+        $this->tmpInputFile = config('ptp.temp_dir').'/'.$inputFile.'.json';
+        $this->tmpImageInputFile = config('ptp.temp_dir').'/'.$inputFile.'_images.json';
+        $this->user = $user;
+        $this->id = $id;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $callback = function ($images, $paths){
+            $this->generateImageInputFile($paths,  $images);
+        };
+        $imageData = $this->generateInputFile();
+        $imageIds = array_keys($imageData);
+        $images = Image::whereIn('id', $imageIds)->get()->all();
+        FileCache::batch($images, $callback);
+        $this->python();
+        $this->uploadConvertedAnnotations();
+        $this->cleanupJob();
+        $this->user->notify(new PtpJobConcluded($this->volumeName));
+    }
+
+    /**
+     * Generate the input Job File
+     *
+     * @return
+     */
+    public function generateInputFile(): array
+    {
+        $imageAnnotationArray = [];
+
+        $pointShapeId = Shape::pointId(); //Find annotations with selected label in desired volume
```

Comment:
This comment does not describe the current line. Generally, describe in a comment *why* something is done instead of *what* (because *what* can be determined from the code). Of course there are exceptions when the code is very complicated. I don't think a comment is needed here.

```suggestion
        $pointShapeId = Shape::pointId();
```